   ml_ops/sm-datazone_import/import-sagemaker-domain.py

```diff
@@ -398,9 +418,27 @@ def _associate_fed_role(self):
             else:
                 print(f"Caught error: {repr(e)}")
 
+    def _cross_account_action(self):
+        # Decision to link domain from X-account. Required to assumeRole from child-account containing SM domain.
+
+        decision = input("Need to import SM profiles from a X-account? [y/n]: ")
+        if decision == "y":
+            x_account_role = input(
+                "X-account role ARN to assume? This would be the account that originally created"
+                "the DataZone domain, aka the parent account: "
+            )
+            session = assumed_role_session(x_account_role)
+            dz_endpoint_url = "https://datazone." + region + ".api.aws"  # prod
+            self.dz_client = session.client(
+                "datazone", region_name=region, endpoint_url=dz_endpoint_url
+            )
+            self.byod_client = session.client(
+                "datazone-byod", region_name=region, endpoint_url=dz_endpoint_url
+            )
```

Comment:
not clear where these are used - looks like this is overwriting the clients - do we need both? (clients for DZ domain account and SM domain account)

Reply:
Yea. These override the existing clients because we need to assume a new role/session. I think we can extend these with a new var so that the original WF isnt touched.